src/mtype.h

```diff
@@ -346,6 +346,9 @@ struct mtype {
         ::weakpoints weakpoints;
         weakpoint_families families;
 
+        // Traps avoided by this monster
+        std::vector<std::string> trap_avoids;
```

Comment:
A set is more convenient for lookup (and faster). Also, use the ids instead of raw strings.
Also, iterate through this in `MonsterGenerator::finalize_mtypes` and check that each `trap_id` `.is_valid()`.
```suggestion
        std::set<trap_id> trap_avoids;
```

Reply:
Resolved